algviz/picture/connection_map.py

```diff
@@ -0,0 +1,136 @@
+            super().__init__(self, start_point, connect_pic, anchor)
+
+        def is_pointer(self):
+            return True
+    
+    def __init__(self, startPicture):
+        self.connections = []
+        self.picture = startPicture
+        
+    def add_connection(self, point, connect_pic, anchor=None, pointer=True, pointer_style={}):
+        if self.picture.width < point[0] or self.picture.height < point[1]:
+            raise TypeError("Point not included in picture")
+        if pointer:
+            self.connections.append(ConnectionMap.PointerConnection(point, connect_pic, anchor, pointer_style))
+        else:
+            self.connections.append(ConnectionMap.Connection(point, connect_pic, anchor))
+
+    def _resize_pic(self):
+        left_width = 0
+        right_width = 0
+        left_height = 0
+        right_height = 0
+        for connection in self.connections:
+            if not connection.is_pointer():
+                #in place connections dont change size of picture
+                continue
+            connect_size = connection.connect_pic.size
+            if connection.anchor == Anchor.LEFT:
+                #left anchor = picture should be placed to right of main picture
+                right_width = max(connect_size[0], right_width)
+                right_height += connect_size[1]
+            elif connection.anchor == Anchor.RIGHT:
+                #right anchor = picture placed on left of main picture
+                left_width = max(connection_size[1], left_width)
+                left_height += connect_size[1]
+            else:
+                #Otherwise, picture connected on whatever side its pointer is closes to
+                #There are better ways to do this, but I wont improve unless graphviz doesn't pan out
+                if connection.start_point[0] <= self.picture_size[0]/2:
+                    left_width = max(connection_size[1], left_width)
+                    left_height += connect_size[1]
+                else:
+                    right_width = max(connect_size[0], right_width)
+                    right_height += connect_size[1]
+        new_width = left_width + self.picture.width + right_width
+        new_height = max(left_height, self.picture.height, right_height)
+        self.picture.width = new_width
+        self.picture.height = new_height
+        self.picture.size = (new_width, new_height) #TODO: remove
+        return left_width, left_height, right_width, right_height
+
+    def _determine_pointer_placement(self, connection, anchor, mid):
+        if anchor == Anchor.LEFT or anchor == Anchor.TOPLEFT:
+            pic_side = "right"
+        elif anchor == Anchor.RIGHT or connection.start_point <= mid:
+            pic_side = "left"
+        else:
+            pic_side = "right"
+                    
+    def draw_connections(self):
+        #TODO: reimplement with GraphViz?
+        start_pic_height = self.picture.width
+        start_pic_width = self.picture.height
```

Comment:
`start_pic_height = self.picture.width
start_pic_width = self.picture.height`

Is this intentional?